### DISCUSSION TITLE
SWF that uses "Import for runtime sharing"

### DISCUSSION THREAD

**Discussion by antartoo** [ruffle-rs/ruffle#15089] [General]
I have some SWFs developed with Flash MX in 2000s that use symbols/MovieClips stored in another shared SWF file in runtime (different from using loadMovie()). 

The UI to define symbol Linkage/runtime sharing in Flash looked like this:
https://ptgmedia.pearsoncmg.com/images/art_hoekman_10Minutes3/elementLinks/fig08.gif

I tested with ruffle player (e.g. file1.swf) it didn't seem to work, with errors like this 
ERROR core/src/library.rs:199 Tried to instantiate non-registered character ID 14
ERROR core/src/display_object/movie_clip.rs:1700 Unable to instantiate display node id 14, reason being: Character id doesn't exist

It seemed the player couldn't find the symbol/movieclip within file1.swf, and the player didn't know the symbol was located in another SWF (e.g. sharedLib.swf) and load that symbol from sharedLib.swf.

Does anyone if this "runtime sharing" in SWF is supported in the ruffle player?

Thanks

**Comment by Lord-McSweeney** [comment]
This is the unimplemented `ImportAssets` tag- we plan to support it, but it has not been implemented yet.

**Comment by antartoo** [comment]
Good to know. Thank you for your reply.
